src/main/kotlin/io/cloudflight/gradle/autoconfigure/java/JavaConfigurePlugin.kt

```diff
@@ -106,5 +120,15 @@ class JavaConfigurePlugin : Plugin<Project> {
         const val EXTENSION_NAME = "javaConfigure"
     }
 
-
+    class JavaAutoConfigurePlugin : Plugin<Project> {
```

Comment:
I'm not sure if that's good - in that single case here it may make sense, but setting conventions could also contain a bit more of business logic, and putting all that stuff together for java, kotlin, swagger, npm,... into the root plugin could end us up with a bunch of mess in the end. We do have all the extension classes on the root - because there is no other option, if we want to have that in one common `autoConfigure` extension, but populating the defaults imho should come from the business logic package